src/app.rs

```diff
@@ -334,5 +372,28 @@ impl App {
                 }
             });
         f.render_widget(canvas, chunks[0]);
+        if self.rosout_widget_enabled {
+            f.render_widget(self.build_rosout_widget(), chunks[1]);
+        }
+    }
+
+    pub fn toggle_rosout_widget(&mut self) {
+        if self.rosout_listener.is_buffering() {
+            self.rosout_listener.toggle_buffering();
+        } else if self.rosout_widget_enabled {
+            self.rosout_widget_enabled = !self.rosout_widget_enabled;
+        } else {
+            self.rosout_listener.toggle_buffering();
+            self.rosout_widget_enabled = true;
+        }
+    }
+
+    pub fn build_rosout_widget(&mut self) -> Paragraph {
+        let mut all_spans = Vec::<Spans>::new();
+        for logstring in self.rosout_listener.read_logstring_buffer().iter() {
+            all_spans.extend(ansi_to_text(logstring.bytes()).unwrap().lines);
```

Comment:
the coloring is not consistent for me.  Why not store the message + level as a struct  in the buffer and use the level set  to set the foreground color on the span?  That would also remove the dependency